src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -977,30 +972,45 @@ int xccdf_policy_generate_fix(struct xccdf_policy *policy, struct xccdf_result *
 
 	// In Ansible we have to generate variables first and then tasks
 	if (strcmp(sys, "urn:xccdf:fix:script:ansible") == 0) {
+		struct oscap_list *variables = oscap_list_new();
+		struct oscap_list *tasks = oscap_list_new();
+		struct oscap_iterator *rules_to_fix_it = oscap_iterator_new(rules_to_fix);
```

Comment:
I propose to do some function extraction here. What about introducing `get_vars_and_tasks_from_rules(rules_to_fix, variables, tasks)` here?